system/lib/wasmfs/file.h

```diff
@@ -255,6 +257,19 @@ class Directory : public File {
     }
   }
 };
+
+struct ParsedPath {
+  std::optional<Directory::Handle> parent;
+  std::shared_ptr<File> child;
+};
+
+// TODO: When locking the directory structure is refactored, parent should be
+// returned as a pointer, similar to child. Given a pathname, this function will
+// return a locked parent directory and a pointer to the specified file.
+ParsedPath getParsedPath(std::vector<std::string> pathParts,
+                         long& err,
+                         std::shared_ptr<File> forbiddenAncestor = nullptr);
+
```

Comment:
The comments below this (explaining `forbiddenAncestor` etc.) look relevant to this function as well.

And, a comment comparing the two functions would be good. Something like, "call foo if you need X, call bar if you also need Y" perhaps, or "this function is similar to the above foo(), except that .."